test/e2e/client_tls_test.go

```diff
@@ -384,6 +313,535 @@ func TestClientTLS(t *testing.T) {
+		// Name is the name of the test case.
+		Name string
+		// CreateCerts generates the certificates for the test case. Certificates and CRLs must not have expired at the
+		// time of the run, so they must be generated at runtime.
+		CreateCerts func() TestMTLSWithCRLsCerts
+	}{
+		{
+			// This test case has CA certificates including a CRL distribution point (CDP) for the CRL that they
+			// generate and sign. This is the default way to distribute CRLs according to RFC-5280
+			Name: "subject-crl",
+			CreateCerts: func() TestMTLSWithCRLsCerts {
+				crlHostURL := crlHostName.Name + "." + crlHostName.Namespace + ".svc"
+				rootCDP := "http://" + crlHostURL + "/root-ca.crl"
+				intermediateCDP := "http://" + crlHostURL + "/intermediate-ca.crl"
+
+				rootCA, err := CreateTLSKeyCert("testing root CA", time.Now(), time.Now().Add(24*time.Hour), true, []string{rootCDP}, nil)
+				if err != nil {
+					t.Fatalf("Failed to create root CA certificate: %v", err)
+				}
+				intermediateCA, err := CreateTLSKeyCert("testing intermediate CA", time.Now(), time.Now().Add(24*time.Hour), true, []string{intermediateCDP}, &rootCA)
+				if err != nil {
+					t.Fatalf("Failed to create intermediate CA certificate: %v", err)
+				}
+
+				signedByRoot, err := CreateTLSKeyCert("client signed by root", time.Now(), time.Now().Add(24*time.Hour), false, nil, &rootCA)
+				if err != nil {
+					t.Fatalf("Failed to create client certificate: %v", err)
+				}
+				signedByRoot.CertPem = strings.Join([]string{signedByRoot.CertPem, rootCA.CertPem}, "\n")
+				signedByIntermediate, err := CreateTLSKeyCert("client signed by intermediate", time.Now(), time.Now().Add(24*time.Hour), false, nil, &intermediateCA)
+				if err != nil {
+					t.Fatalf("Failed to create client certificate: %v", err)
+				}
+				signedByIntermediate.CertPem = strings.Join([]string{signedByIntermediate.CertPem, intermediateCA.CertPem, rootCA.CertPem}, "\n")
+				revokedByRoot, err := CreateTLSKeyCert("client revoked by root", time.Now(), time.Now().Add(24*time.Hour), false, nil, &rootCA)
+				if err != nil {
+					t.Fatalf("Failed to create client certificate: %v", err)
+				}
+				revokedByRoot.CertPem = strings.Join([]string{revokedByRoot.CertPem, rootCA.CertPem}, "\n")
+				revokedByIntermediate, err := CreateTLSKeyCert("client revoked by intermediate", time.Now(), time.Now().Add(24*time.Hour), false, nil, &intermediateCA)
+				if err != nil {
+					t.Fatalf("Failed to create client certificate: %v", err)
+				}
+				revokedByIntermediate.CertPem = strings.Join([]string{revokedByIntermediate.CertPem, intermediateCA.CertPem, rootCA.CertPem}, "\n")
+				selfSigned, err := CreateTLSKeyCert("self signed cert", time.Now(), time.Now().Add(24*time.Hour), false, nil, nil)
+				if err != nil {
+					t.Fatalf("Failed to create client certificate: %v", err)
+				}
+
+				_, rootCRLPem, err := CreateCRL(nil, rootCA, time.Now(), time.Now().Add(10*time.Minute), RevokeCertificates(time.Now(), revokedByRoot))
```

Comment:
Anything significant about this 10 min next update value? Is there any chance that these tests could take 10 mins and something would break? Should it be like an hour to be safe? Does it matter?